lib/src/accordion/accordion.component.ts

```diff
@@ -0,0 +1,175 @@
+    iconRotation?: AccordionIconRotation;
+    id?: string;
+    list: Array<AccrodionListItem>;
+    alternative?: boolean;
+    activeIndex?: number;
+}
+
+@Component({
+    selector: "sebng-accordion",
+    templateUrl: "./accordion.component.html",
+    styleUrls: ["./accordion.component.scss"],
+    encapsulation: ViewEncapsulation.None,
+})
+export class AccordionComponent implements OnInit, OnChanges, AfterViewInit, AfterViewChecked {
+    @Input() className?: string;
+    @Input() customIcon?: string;
+    @Input() customIconExpanded?: string;
+    @Input() iconPosition?: "left" | "right";
+    @Input() iconRotation?: AccordionIconRotation;
+    @Input() id?: string;
+    @Input() list: Array<AccrodionListItem>;
+    @Input() alternative?: boolean;
+    @Input() activeIndex?: number;
+
+    @ViewChildren("accordionItemRefs") accordionItemRefs: QueryList<ElementRef>;
+
+    public idList: Array<string>;
+    public accordionClassName: string = "custom-accordion";
+    public itemClassName: string = "custom-accordion";
+    public active: number = null;
+
+    public heightList: Array<number>;
+
+    constructor(private changeDetector: ChangeDetectorRef) {}
+
+    //helper functions
+    constructIds() {
+        this.idList = this.list.map(() => randomId("accordion-"));
+    }
+
+    /** Constructs the `className` to be used in accordion wrapper */
+    constructClassName(): void {
+        let cn: string = "custom-accordion";
+        cn += this.className ? ` ${this.className}` : "";
+        cn += this.alternative ? " alternative-accordion" : "";
+        this.accordionClassName = cn;
+    }
+
+    /** Constructs the `className` to be used in accordion items */
+    constructItemClassName(): void {
+        let cn: string = "accordion-item";
+        cn += " " + (this.iconPosition ? this.iconPosition : "left");
+        cn += " " + (this.iconRotation ? this.iconRotation : "deg-180");
+        cn += this.customIconExpanded ? " transform" : "";
+        this.itemClassName = cn;
+    }
+
+    expandOrCollapseSection(itemIndex: number): void {
+        const updatedHeightList: Array<number> = Array(this.list?.length).fill(0);
+        updatedHeightList[itemIndex] = this.heightList[itemIndex]
+            ? 0
+            : this.accordionItemRefs?.toArray()[itemIndex]?.nativeElement?.scrollHeight;
+        this.heightList = updatedHeightList;
+    }
+
+    toggle(index: number): void {
+        if (this.active === index) {
+            // Section already expanded
+            this.expandOrCollapseSection(index);
+        } else {
+            if (this.active !== null) {
+                // Another section is already expanded
+                this.expandOrCollapseSection(this.active);
+            }
+            this.expandOrCollapseSection(index);
+        }
+        this.active = this.active === index ? null : index;
+    }
+
+    contentIsOfTypeArray(content: AccordionContentType): boolean {
+        return content instanceof Array;
```

Comment:
I believe this is more accurate, you can read more about the difference [here](https://stackoverflow.com/questions/22289727/difference-between-using-array-isarray-and-instanceof-array) 
```suggestion
        return Array.isArray(content);
```

Reply:
isArray support wasn't that good before. Unless if its updated, i tend to avoid it.